templates/partial/count.html.twig

```diff
@@ -0,0 +1,7 @@
+{% set i =  (index.getCurrentPageNumber - 1) * index.getItemNumberPerPage %}
+
+{% if index.getTotalItemCount > index.getItemNumberPerPage %}
+
+<p class="count">Displaying {{ i + 1}}–{{ min(index.getCurrentPageNumber * index.getItemNumberPerPage, index.getTotalItemCount) }} {{ label }} out of {{ index.getTotalItemCount }} total</p>
+
+{% endif %}
```

Comment:
You can simplify the pagination you’re doing a fair bit. The pagination bundle already does the arithmetic you’re doing. Here’s a possibility:

```twig
{% set pagination = index.paginationData %}
{% if index.getTotalItemCount > index.getItemNumberPerPage %}

<p class="count">Displaying {{ pagination.firstItemNumber }}–{{ pagination.lastItemNumber }}
    {{ label }} out of {{ index.getTotalItemCount }} total</p>

{% endif %}
```

The data available in `pagination` is 

Last, current, numItemsPerPage, first, pageCount, totalCount, pageRange, startPage, endPage, previous, next, pagesInRange, firstPageInRange, lastPageInRange, currentItemCount, firstItemNumber, and lastItemNumber.

Reply:
Ah! Much nicer. I though I had figured out all of the available data for pagination, but I hadn't--that's good to know, thanks! I've added this now, except I got rid of the conditional, since we should probably show how many things are there even if it doesn't cross pages.